docs/modules/serialization/pages/serialization.adoc

```diff
@@ -1,32 +1,75 @@
 = Serialization
+:description: To be able to send data over a network between cluster members and/or clients, you have to serialize it into raw bytes. Hazelcast has many serialization methods to choose from, depending on which Hazelcast topology you are using and what you plan on doing with your data.
 
-Hazelcast needs to serialize the Java objects that you put into Hazelcast because
-Hazelcast is a distributed system. The data and its replicas are
-stored in different partitions on multiple cluster members.
-The data you need may not be present on the local member, and in that case,
-Hazelcast retrieves that data from another member. This requires serialization.
+{description}
 
-Serialization is used in the following cases:
+== Serialization in Client/Server Mode
 
-* Adding key/value objects to a map
-* Putting items in a queue/set/list
-* Sending a runnable using an executor service
-* Processing an entry within a map
-* Locking an object
-* Sending a message to a topic
+In xref:overview:topology.adoc[client/server mode], you can serialize your data on the client side and/or the xref:ROOT:glossary.adoc#member[member] (server) side.
 
-Hazelcast optimizes the serialization for the basic types and their array types
-with its built-in serializers. By default, you cannot override this behavior; however,
-Hazelcast offers a way to override them using a configuration option,
-see the xref:overriding-built-in-serializers.adoc[Overriding the Default Serializers section].
-
-Hazelcast's built-in serializers can handle the following types:
+If you just want to store data on members, clients can handle serialization without the members needing to know how to do so. For example, you can serialize data as JSON and add it to a map. Then you can later request that information and deserialize on the client.
+
+However, in the following cases, you must configure members to be able to serialize and deserialize your data:
+
+[cols="a,a,a,a"]
+|===
+|Case|Clients|Reason|Exceptions
+
+|When the data structure's values are xref:data-structures:map.adoc#setting-in-memory-format[stored as objects]
+|All clients
+|Members must be able to serialize or deserialize the object in order to send it to other members or receive it from other members
+|
+
+|When you want to xref:computing:entry-processor.adoc[process map entries] or use an xref:computing:executor-service.adoc[executor service]
+|All clients
+|Members need to serialize the tasks
+|When using the Java client with xref:implementing-portable-serialization.adoc[`Portable`] serialization, you don't need to configure serialization on the member. For details, see xref:clusters:accessing-domain-objects.adoc[].

```

Comment:
For the EntryProcessor and ExecutorService, `Portable` is not enough. User has to configure serialization on the members. Users implementation of EntryProcessor, Runnable,Callable has to be in the members classpath(or should be deployed via user-code-deployment)

But for the entries(key,value of the map) that EntryProcessor accesses, being Portable is enough. Those are the domain objects and reference to `xref:clusters:accessing-domain-objects.adoc` make sense for those.

Reply:
So we may want to split this part to two may be to differentiate domain objects from EntryProcessor class itself(ExecutorService also falls in to EntryProcessor section)

Reply:
So we can change the exception to:

When using the Java client, you can configure map entries with xref:implementing-portable-serialization.adoc[`Portable`] serialization, in which case you don't need to configure serialization on the member. For details, see xref:clusters:accessing-domain-objects.adoc[].